dunning_block/models/account_move.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2022 Solvti (http://www.solvti.pl).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import datetime
+import logging
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    dunning_block_active = fields.Boolean(
+        string="Dunning Block",
+        default=False,
+        readonly=True,
+        store=True,
+        tracking=True,
+        copy=False,
+    )
+    dunning_block_end_date = fields.Date(
+        readonly=True, store=True, tracking=True, copy=False
+    )
+    dunning_block_note = fields.Text(
+        readonly=True, store=True, tracking=True, copy=False
+    )
+    dunning_block_history_ids = fields.One2many(
+        "dunning.block.history", "move_id", readonly=True, copy=False
+    )
+    dunning_block_days_left = fields.Integer(compute="_compute_dunning_block_days_left")
+    dunning_block_reason_id = fields.Many2one(
+        "dunning.block.reason", readonly=True, store=True, tracking=True, copy=False
+    )
+
```

Comment:
Instead of overloading the stock.move model it would be better to use dunning.block.history (or another similar model) to store the blocking information and query directly from that list which invoices should be dunned.

Reply:
Totally agree, would be better to have it another model and maybe just link it with singe many2one field. I could maybe rewrite it if there would be an interest in this functionality